src/core/presentation/mse.js

```diff
@@ -102,8 +102,17 @@ export class MSEBuffer {
             this.cleaning = true;
             promises.push(new Promise((resolve, reject)=>{
                 this.cleanResolvers.push(resolve);
-                this.sourceBuffer.remove(this.sourceBuffer.buffered.start(i), this.sourceBuffer.buffered.end(i));
-                resolve();
+                if (!this.sourceBuffer.updating) {
+                    this.sourceBuffer.remove(this.sourceBuffer.buffered.start(i), this.sourceBuffer.buffered.end(i));
```

Comment:
hi @mburger81 

why we should remove the sourceBuffer? sourceBuffer size limit ?